lib/rules/prefer-template.js

```diff
@@ -187,6 +187,24 @@ module.exports = {
                 return sourceCode.getText(currentNode);
             }
 
+            if (isConcatenation(currentNode) && !hasNonStringLiteral(currentNode)) {
+                const firstToken = sourceCode.getFirstToken(currentNode.left);
+                const currentNodeText = sourceCode.getText(currentNode);
+                const parentNodeText = sourceCode.getText(currentNode.parent);
+
+                if (parentNodeText.startsWith(currentNodeText)) {
+                    const leftNodeText = currentNodeText.slice(0, currentNode.right.range[0] - firstToken.range[0]);
+
+                    return `${leftNodeText}\`${currentNode.right.value}'`;
```

Comment:
This converts the trailing string literal to a template element without escaping (which would be done in the conditional that is on lines 168-184 if this was a lone string literal).

```js
"a" + "`b`" + c;
```

autofixed to:

```js
"a" + ``b`${  c}`;
```

Reply:
Also, `currentNode.right` doesn't have to be a string literal, it could be a parenthesized concatenation of string literals:

```js
"a" + ("b" + "c") + d;
```

autofixed to:

```js
"a" + (`undefined${  d}`;
```

Reply:
Updated the code and added more test cases 👍🏻